opentasks-provider/src/main/java/org/dmfs/provider/tasks/processors/instances/Validating.java

```diff
@@ -43,50 +69,79 @@ public Validating(EntityProcessor<InstanceAdapter> delegate)
     public InstanceAdapter insert(SQLiteDatabase db, InstanceAdapter entityAdapter, boolean isSyncAdapter)
     {
         validate(entityAdapter, isSyncAdapter);
-        return mDelegate.insert(db, entityAdapter, isSyncAdapter);
+
+        Optional<Long> instanceId = new NullSafe<>(entityAdapter.taskAdapter().valueOf(TaskAdapter.ORIGINAL_INSTANCE_ID));
+        if (instanceId.isPresent())
+        {
+            String timeStampString = Long.toString(entityAdapter.taskAdapter().valueOf(TaskAdapter.ORIGINAL_INSTANCE_TIME).getTimestamp());
+            // there better be no instance for this yet, otherwise this is going to fail
```

Comment:
I am not sure I understand the "better be" phrasing here, it sounds like hoping for, not sure. 
Could this be stated as well?: "If there is an instance already for this id and time, this is going to throw."

Reply:
updated